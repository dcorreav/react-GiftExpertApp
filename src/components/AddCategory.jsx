import { useState } from "react"

import { PropTypes } from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

    // antes estaba setCategories

    const [InputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {        

        event.preventDefault();

        if (InputValue.trim().length <= 1) return;

        //setCategories(categories => [InputValue, ...categories])

        setInputValue('');
        onNewCategory(InputValue.trim());
        
    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label="form" >
            <input
                type="text"
                placeholder="Buscar Gifts"
                value={InputValue}
                onChange={onInputChange}
            ></input>
        </form>


    )


}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}