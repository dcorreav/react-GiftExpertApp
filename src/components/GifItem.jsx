import PropTyes from 'prop-types';

export const GifItem = ({ title, url }) => {
    return (

        <div className="card">
            <img src={url} alt={title} ></img>
            <p> {title} </p>

        </div>

    )
}



GifItem.PropTyes = {
    title: PropTyes.string.isRequired,
    url: PropTyes.string.isRequired,
}