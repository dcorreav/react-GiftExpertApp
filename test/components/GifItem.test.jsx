

import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/components/GifItem';




describe('pruebas sobre el componente GifItem', () => {

    const title = 'Sayonara';
    const url = 'https://sayprana.com/sayonara.jpg';



    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} ></GifItem>)

        expect(container).toMatchSnapshot();

    });


    test('debe mostrar la imagen con el url y el alt indicado', () => {

        render(<GifItem title={title} url={url} ></GifItem>)

        //screen.debug();

        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);

        //console.log(screen.getByRole('img').src);

    })



    test('debe de mostrar el titulo en el componente', () => {


        render(<GifItem title={title} url={url} ></GifItem>)

        expect( screen.getByText(title)).toBeTruthy();



    })


})