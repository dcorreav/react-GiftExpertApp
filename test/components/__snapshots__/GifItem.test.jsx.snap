// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sayonara"
      src="https://sayprana.com/sayonara.jpg"
    />
    <p>
       
      Sayonara
       
    </p>
  </div>
</div>
`;
